src/utils/convert-utils.ts

```diff
@@ -298,3 +301,102 @@ export const getCurveAmountOut = async (
     amountOutWithoutFees
   };
 };
+
+type PoolPairBase = {
+  id: string;
+  poolType: BalancerPoolType;
+  address: string;
+  swapFee: number;
+  tokenIn: string;
+  tokenOut: string;
+  decimalsIn: number;
+  decimalsOut: number;
+  balanceIn: number;
+  balanceOut: number;
+};
+
+type WeightedPoolPairData = PoolPairBase & {
+  weightIn: number;
+  weightOut: number;
+};
+
+const getPoolPairData = async (
+  network: Network,
+  poolId: string,
+  tokenIn: string,
+  tokenOut: string
+): Promise<WeightedPoolPairData> => {
+  if (network !== "arbitrum") throw new Error("can only get pool pair data on arbitrum");
+  const graphUrl = config.theGraphEndpoints.arbitrum.balancer;
+
+  const { pool } = await request(
+    graphUrl,
+    gql`
+      query {
+        pool(id:"${poolId.toLowerCase()}") {
+          id
+          address
+          poolType
+          factory
+          swapFee
+          tokens(first:1000) {
+            address
+            decimals
+            balance
+            weight
+          }
+        }
+      }
+  `
+  );
+
+  const tokenInResponse = pool.tokens.find(
+    (token: any) => token.address.toLowerCase() === tokenIn.toLowerCase()
+  );
+  const tokenOutResponse = pool.tokens.find(
+    (token: any) => token.address.toLowerCase() === tokenOut.toLowerCase()
+  );
+
+  mustExist(tokenInResponse, `token in not in pool with id ${poolId}`);
+  mustExist(tokenOutResponse, `token out not in pool with id ${poolId}`);

```

Comment:
```suggestion
  const tokenInResponse = mustExist(
    pool.tokens.find(
      (token: any) => token.address.toLowerCase() === tokenIn.toLowerCase()
    ),
    `token in not in pool with id ${poolId}`
  );
  const tokenOutResponse = mustExist(
    pool.tokens.find(
      (token: any) => token.address.toLowerCase() === tokenOut.toLowerCase()
    ),
    `token out not in pool with id ${poolId}`
  );
```
I think it would be better to either use the return value of `mustExist` or just do an `if (!tokeninResponse) throw new Error(...`.